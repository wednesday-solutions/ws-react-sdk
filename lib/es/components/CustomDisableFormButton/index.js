import{_ as n,a as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/button/style";import t from"antd/lib/button";import"antd/lib/form/style";import e from"antd/lib/form";import"../_rollup-plugin-inject-process-env-1259e43d.js";import o from"react";import i from"prop-types";import a from"styled-components";import{c as m}from"../colors-93a064d8.js";import{f as l}from"../fonts-ca6864a0.js";import{m as s}from"../media-ee9cbc51.js";var p,d,f=a(e.Item)(p||(p=n(["\n  && {\n    .ant-form-item-explain,\n    .ant-form-item-extra {\n      text-align: left;\n    }\n    .ant-form-item {\n      margin: 0;\n    }\n    .ant-form-item-has-error .ant-input,\n    .ant-form-item-has-error .ant-input:hover,\n    .ant-form-item-has-error .ant-input,\n    .ant-form-item-has-error .ant-input:focus-within {\n      background-color: ",";\n      box-shadow: none;\n    }\n    .ant-form-item-has-error .ant-input-affix-wrapper,\n    .ant-form-item-has-error .ant-input-affix-wrapper:hover {\n      background-color: ",";\n      box-shadow: none;\n    }\n    &.ant-form-item-control-input {\n      min-height: 3.5rem;\n      ","\n      ","\n    width: 100%;\n    }\n  }\n"])),m.white,m.white,s.desktop.max("\n      min-height: 3rem;\n    "),s.largeMobile.max("\n      min-height: 2.5rem;\n    ")),h=a(t)(d||(d=n(["\n  &.ant-btn-lg {\n    ",";\n    border-radius: 8px;\n    height: 3.5rem;\n    ","\n    ","\n    ",";\n    margin-bottom: 1rem;\n    border: 2px solid ",";\n  }\n"])),(n=>n.width&&"width: ".concat(n.width,"%;")),s.desktop.max("\n      height: 3rem;\n    "),s.largeMobile.max("\n      height: 2.5rem;\n    "),l.style.xRegularFW600(),m.green);function u(n){return o.createElement(f,{"data-testid":"custom-disable-form-button",shouldUpdate:!0},(()=>{return o.createElement(h,r({disabled:!(null!=n&&null!==(e=n.form)&&void 0!==e&&e.isFieldsTouched(!0)&&(null==n||null===(t=n.form)||void 0===t||!t.getFieldsError().filter((n=>{var{errors:r}=n;return r.length})).length))},n),n.text);var t,e}))}u.propTypes={isFieldsTouched:i.func,getFieldsError:i.func,text:i.string,form:i.object};export default u;
