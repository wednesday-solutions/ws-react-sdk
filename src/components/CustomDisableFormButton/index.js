/**
 *
 * CustomDisableFormButton
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Form, Button } from 'antd';
import styled from 'styled-components';
import { media, colors, fonts } from '@themes';

const DFI = styled(Form.Item)`
  && {
    .ant-form-item-explain,
    .ant-form-item-extra {
      text-align: left;
    }
    .ant-form-item {
      margin: 0;
    }
    .ant-form-item-has-error .ant-input,
    .ant-form-item-has-error .ant-input:hover,
    .ant-form-item-has-error .ant-input,
    .ant-form-item-has-error .ant-input:focus-within {
      background-color: ${colors.white};
      box-shadow: none;
    }
    .ant-form-item-has-error .ant-input-affix-wrapper,
    .ant-form-item-has-error .ant-input-affix-wrapper:hover {
      background-color: ${colors.white};
      box-shadow: none;
    }
    &.ant-form-item-control-input {
      min-height: 3.5rem;
      ${media.desktop.max(`
      min-height: 3rem;
    `)}
      ${media.largeMobile.max(`
      min-height: 2.5rem;
    `)}
    width: 100%;
    }
  }
`;

const DCB = styled(Button)`
  &.ant-btn-lg {
    ${props => props.width && `width: ${props.width}%;`};
    border-radius: 8px;
    height: 3.5rem;
    ${media.desktop.max(`
      height: 3rem;
    `)}
    ${media.largeMobile.max(`
      height: 2.5rem;
    `)}
    ${fonts.style.xRegularFW600()};
    margin-bottom: 1rem;
    border: 2px solid ${colors.green};
  }
`;

function CustomDisableFormButton(props) {
  const isFieldsTouchedCheck = () => !props?.form?.isFieldsTouched(true);

  const hasError = () => !!props?.form?.getFieldsError().filter(({ errors }) => errors.length).length;

  return (
    <DFI data-testid="custom-disable-form-button" shouldUpdate={true}>
      {() => (
        <DCB disabled={isFieldsTouchedCheck() || hasError()} {...props}>
          {props.text}
        </DCB>
      )}
    </DFI>
  );
}

CustomDisableFormButton.propTypes = {
  isFieldsTouched: PropTypes.func,
  getFieldsError: PropTypes.func,
  text: PropTypes.string,
  form: PropTypes.object
};

export default CustomDisableFormButton;
