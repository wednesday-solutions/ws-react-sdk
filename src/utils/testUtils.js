import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
