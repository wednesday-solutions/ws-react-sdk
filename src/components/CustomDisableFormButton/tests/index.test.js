/**
 *
 * Tests for CustomDisableFormButton
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import CustomDisableFormButton from '../index';

describe('<CustomDisableFormButton />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomDisableFormButton />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomDisableFormButton component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomDisableFormButton />);
    expect(getAllByTestId('custom-disable-form-button').length).toBe(1);
  });
});
