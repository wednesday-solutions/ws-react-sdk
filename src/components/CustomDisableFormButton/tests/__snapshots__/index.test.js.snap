// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomDisableFormButton /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-row ant-form-item sc-bdfBwQ cUsKxY"
      data-testid="custom-disable-form-button"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn sc-gsTCUz hmKVcQ"
              disabled=""
              type="button"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
